Update episode 08-development-setup

SarahAlidoost opened the pull request:
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [x] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #167 #164
Follow up issue #190

SarahAlidoost commented:
The remaining CI errors are related to the links at the end of the file and the `markdown` code blocks. I couldn't find a better-supported language by the template, see the list [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/ff50d10915459a51336bfdc94716b46cffbdbd4a/bin/lesson_check.py#L73).

Peter9192 reviewed (changes requested):
Hey @SarahAlidoost, nice work! It's become much more extensive than I initially expected, but I think many people are gonna benefit from it. Since it covers quite a lot of topics, I have quite a few comments... but I tried to provide suggestions whereever I could :-)

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+After the authentication, a folder called ``ESMValTool`` is created in your working directory.
+This folder contains the source code of the tool.
+To continue the installation, we move into the ``ESMValTool`` directory
+and check out the ``master`` branch:
```
correct, it is replaced by a note that it is the default.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+~~~bash
+cd ESMValTool
+git checkout master
+~~~
```
I added the output of git clone and git status. Please let me know if it is not what you meant.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+> first delete it or choose another name for the new environment in this lesson.
+> For more information see
+> [conda managing environments][manage-environments].
+{: .callout}
```
Agree, I added your suggestion.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
-If you run into trouble, please try recreating the environment. More information
-can be found in the [conda documentation](https://docs.conda.io/en/latest/).
+If installation is successful, ESMValTool prints a help message to the console.
+For more details about development installation, see ESMValTool documentation on
+[install from source][install-from-source].
```
Agree, I remove the link to the documentation from here.

Peter9192 replied:
It's still there, right?

SarahAlidoost replied:
oops, my bad, it is removed now.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+- helping  with reviewing process of pull requests, see ESMValTool documentation on
+[Review of pull requests](https://docs.esmvaltool.org/en/latest/community/review.html)
+
+The next sections will explore the ways we can achieve this.
```
Agree, the paragraph is moved to the top. I also added a paragraph similar to what you suggested.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+The pull request will be tested, discussed and merged. This is called "**review process**".
+The process will take some effort and time to learn.
+However, a few “tools” i.e. command lines gets you a long way,
+and we’ll cover those essentials in the next sections.
```
Nice suggestion! I added a tip similar to what you said here.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+As an example, ``Pre-commit`` is a tool that checks your code for any invalid syntax
+and formatting errors. It also fixes some of those errors.
+To explore other tools, have a look at ESMValTool documentation on
+[Code style](https://docs.esmvaltool.org/en/latest/community/introduction.html#code-style).
```
I added the list of the tools as you suggested. However, I didn't remove the link to the documentation as there is some info in the documentation that is beyond the scope of this lesson. Also, I think neither documentation nor this tutorial should cover details about these tools. Because they are related to best practices and code refinery lessons. We might just point users to those resources. This is also in line with what you mentioned about duplicate information.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
-pip install --editable '.[develop]'
+conda activate esmvaltool
+cd ESMValTool
+git checkout your_branch_name
```
right, they are removed.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+
+Previous section introduced some tools to check code style and quality.
+What it hasen’t done is show us how to tell whether our code is getting the right answer.
+To achieve that, we can run tests using ``pytest`` locally:
```
It is added as you suggested.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+>>~~~
+>>
+>> It seems that we need to set a correct path for the diagnostic script
+>> ``warming_stripes.py`` in our recipe!
```
To address this comment, I added a background section in the contribution, and revise the text and exercises. Please have a look again and let me know if it is still confusing.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
 
-The next step is to check that the installation works properly.
-To do this, run the tool with:
+Documentations are available on
```
An introductory sentence is now added.

Peter9192 replied:
```suggestion
The ESMValTool documentation is available on
```
I think words like documentation and codes are usually 'mass nouns' or 'uncountable nouns', so I would avoid using plural forms like 'codes' or 'documentations'. Just like 'water'.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
-esmvaltool --help
+conda activate esmvaltool
+cd ESMValTool
+git checkout your_branch_name
```
Absolutely right. It is now removed.

SarahAlidoost commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,394 @@
+> Add a reference i.e. ``.. _recipe_warming_stripes:``, a section title
+> and some text about the recipe like:
+>
+> ~~~markdown
```
I think `markdown` is one of the languages supported for the code block. However, our template does not support too many languages see [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/ff50d10915459a51336bfdc94716b46cffbdbd4a/bin/lesson_check.py#L73) . I replaced `~~~ `as you suggested. The test is passed but it does not render nicely as before.

SarahAlidoost commented:
> Hey @SarahAlidoost, nice work! It's become much more extensive than I initially expected, but I think many people are gonna benefit from it. Since it covers quite a lot of topics, I have quite a few comments... but I tried to provide suggestions whereever I could :-)

Thank you for reviewing and helpful suggestions. I addressed your comments. Please have another look.

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+> the ``esmvaltool`` environment.
+> Use this command to check that ESMValTool is installed in a ``develop`` mode.
+>
+> **Tips**: see the
```
```suggestion
> **Tip**: see the
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+However, a few “tools” i.e. command lines gets you a long way,
+and we’ll cover those essentials in the next sections.
+
+**Tips**: we encourage you to keep the pull requests small.
```
```suggestion
**Tip**: we encourage you to keep the pull requests small.
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+[recipe_warming_stripes.yml](../files/recipe_warming_stripes.yml)
+to their locations in ESMValTool directory.
+After these changes, you should also check if everthing works fine.
+This is where we take advantage of the tools that are introduced later.
```
Just a development tip from my side: I'm using the 'rewrap' extension in vscode. This let's you auto-wrap lines with `alt+Q` so you don't exceed the line length. I find that super useful for these kind of text blocks.

SarahAlidoost replied:
I just tried this extension, very cool, thank you for the suggestion.

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+
+The pull request will be tested, discussed and merged. This is called "**review process**".
+The process will take some effort and time to learn.
+However, a few “tools” i.e. command lines gets you a long way,
+and we’ll cover those essentials in the next sections.
```
This sentence is not grammatically correct, can you rephrase it? Suggestion: 
```suggestion
However, a few (command line) tools can get you a long way,
and we’ll cover those essentials in the next sections.
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
 
-The next step is to check that the installation works properly.
-To do this, run the tool with:
+- flake8 for checking against the PEP8 style guide
+- yapf to ensure consistent formatting for the whole project
+- isort to consistently sort the import statements
+- yamllint to ensure there are no syntax errors in our recipes and config files
+- lintr for diagnostic scripts written in R
+- codespell to check grammar
```
Can we make those bold or link to their docs?

```suggestion
- [flake8](https://flake8.pycqa.org/en/latest/#) for checking against the PEP8 style guide
- [yapf](https://pypi.org/project/yapf/) to ensure consistent formatting for the whole project
- [isort](https://pypi.org/project/isort/) to consistently sort the import statements
- [yamllint](https://yamllint.readthedocs.io/en/stable/#) to ensure there are no syntax errors in our recipes and config files
- [lintr](https://github.com/jimhester/lintr) for diagnostic scripts written in R
- [codespell](https://pypi.org/project/codespell/) to check grammar
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+### Run unit tests
+
+Previous section introduced some tools to check code style and quality.
+What it hasen’t done is show us how to tell whether our code is getting the right answer.
```
```suggestion
What it hasn’t done is show us how to tell whether our code is getting the right answer.
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+>> ~~~
+>>
+>> For details, see lesson
+>> [Writing your own diagnostic script].
```
Maybe add a dead link for now?
```suggestion
>> [Writing your own diagnostic script](#).
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
 
-For a more complete installation verification, run the automated tests and
-confirm that no errors are reported:
+To build documentations locally, first we make sure that the working directory is ``ESMValTool``
```
```suggestion
To build documentation locally, first we make sure that the working directory is ``ESMValTool``
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
 ~~~
 
-> ## ESMValCore
+Similar to codes, documentations should be well written and adhere to standards.
```
```suggestion
Similar to code, documentation should be well written and adhere to standards.
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+Similar to codes, documentations should be well written and adhere to standards.
+If the documentation is built properly, the previous command prints a message to the console:
+
+~~~bash
+build succeeded.
+
+The HTML pages are in doc/sphinx/build/html.
+~~~
```
```suggestion
~~~
build succeeded.

The HTML pages are in doc/sphinx/build/html.
~~~
{: .output}
```

Peter9192 commented on _episodes/08-development-setup.md:
```diff
@@ -1,138 +1,461 @@
+>> ~~~
+>>
+> {: .solution}
+{: .challenge}
```
A closing remark would be nice. Something like

```suggestion
{: .challenge}

Congratulations! You are now ready to make a pull request.
```

Peter9192 reviewed (changes requested):
Hey Sarah, thanks for making these changes! I have a couple more small suggestions, mostly typos. Let me know when you're ready, then I'll approve the PR.

SarahAlidoost commented:
> Hey Sarah, thanks for making these changes! I have a couple more small suggestions, mostly typos. Let me know when you're ready, then I'll approve the PR.

thank you, your comments are addressed.

SarahAlidoost commented:
Hi @Peter9192 thank you again for reviewing and comments. As we discussed, the failing github action will be fixed in a separate PR. It is mentioned in this issue #191.
